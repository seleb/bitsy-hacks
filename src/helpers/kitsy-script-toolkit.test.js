import {
	convertDialogTags
} from './kitsy-script-toolkit';

describe('convertDialogTags', ()=>{
	it('requires an input string and an input tag', ()=>{
		expect(()=>convertDialogTags()).toThrow();
		expect(()=>convertDialogTags('', '')).not.toThrow();
	});

	it('replaces `(tag "input")` with `{tag "input"}`', ()=>{
		expect(convertDialogTags(`(tag "input")`, 'tag')).toMatchSnapshot();
	});

	it('replaces `(tag input)` with `{tag input}`', ()=>{
		expect(convertDialogTags(`(tag input)`, 'tag')).toMatchSnapshot();
	});

	it('replaces `\\(tag "input"\\)` with `(tag "input")`', ()=>{
		expect(convertDialogTags(`\(tag "input"\)`, 'tag')).toMatchSnapshot();
	});

	it('replaces `\\(tag input\\)` with `(tag input)`', ()=>{
		expect(convertDialogTags(`\(tag input\)`, 'tag')).toMatchSnapshot();
	});

	it('doesn\'t work when nested', ()=>{
		expect(convertDialogTags(`(js "(js "(js "")")")`, 'js')).toMatchSnapshot();
	});

	it('handles relatively complex cases', ()=>{
		// TODO: add some real-world gamedata examples
		expect(convertDialogTags(`a(js "console.log('a');")b{js "console.log('b');"}c(js "console.log('c');")d{js "console.log('d');"}e(js e)f{js f}e(js e)f{js f}`, 'js')).toMatchSnapshot();
	});
})