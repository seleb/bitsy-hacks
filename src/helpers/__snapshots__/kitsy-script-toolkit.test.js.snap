// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertDialogTags doesn't work when nested 1`] = `"{js \\"(js \\"(js \\"\\"}\\")\\")"`;

exports[`convertDialogTags handles relatively complex cases 1`] = `"a{js \\"console.log('a');\\"}b{js \\"console.log('b');\\"}c{js \\"console.log('c');\\"}d{js \\"console.log('d');\\"}e{js e}f{js f}e{js e}f{js f}"`;

exports[`convertDialogTags replaces \`(tag "input")\` with \`{tag "input"}\` 1`] = `"{tag \\"input\\"}"`;

exports[`convertDialogTags replaces \`(tag input)\` with \`{tag input}\` 1`] = `"{tag input}"`;

exports[`convertDialogTags replaces \`\\(tag "input"\\)\` with \`(tag "input")\` 1`] = `"{tag \\"input\\"}"`;

exports[`convertDialogTags replaces \`\\(tag input\\)\` with \`(tag input)\` 1`] = `"{tag input}"`;
